protocols/migration/migration_reports_5_0_0_to_reports_4_0_0.py

```diff
@@ -0,0 +1,175 @@
+import logging
+
+from protocols import reports_4_0_0 as reports_4_0_0
+from protocols import reports_5_0_0 as reports_5_0_0
+from protocols.migration.base_migration import BaseMigration
+from protocols.migration.base_migration import MigrationError
+
+
+class MigrateReports500To400(BaseMigration):
+
+    old_model = reports_5_0_0
+    new_model = reports_4_0_0
+
+    def migrate_clinical_report_rd(self, old_instance):
+        """
+        :type old_instance: reports_5_0_0.ClinicalReportRD
+        :rtype: reports_4_0_0.ClinicalReportRD
+        """
+        new_instance = self.convert_class(self.new_model.ClinicalReportRD, old_instance)  # :type self.new_model.ClinicalReportRD
+
+        # type of interpretationRequestVersion has been changed from int to str
+        new_instance.interpretationRequestVersion = str(old_instance.interpretationRequestVersion)
+
+        # references has been renamed to supportingEvidence
+        new_instance.supportingEvidence = old_instance.references
+
+        # converts all reported variants
+        if old_instance.variants:
```

Comment:
this should be `if old_instance.variants is None:` otherwise we are translating `[]` into `None`

Reply:
:+1:

Reply:
Nope, @priesgo that's not what happens, an empty list and None will evaluate as False in this check:
```
In [1]: if [] :
   ...:     print True
   ...: else:
   ...:     print False
   ...:     
False
```

Reply:
That is precisely what I mean @greglever
let's suppose `old_instance.variants = []`
then you do the following
```
if old_instance_variants:
   new_instance.candidateVariants = [ ... some list comprehension ... ]
```
and you end up having `new_instance.candidateVariants` not being set to `[]` or in other other words we are translating `[]` into `None`.